### PR TITLE
I added swagger config - keep sync with repo :D.

### PR DESCRIPTION
**Pull request by IrinaGlodeanu**
Cleaned controller. Moved logic to service. Don't use action name in URL.

### CODE REVIEW

File: mycinema/pom.xml [hunk 1]
```diff
@@ -42,6 +42,16 @@
             <artifactId>postgresql</artifactId>
             <version>9.4-1206-jdbc42</version>
         </dependency>
+        <dependency>
```
Comment (IrinaGlodeanu) [inline mycinema/pom.xml:45]: added swagger dependency

File: mycinema/src/main/java/ro/fiipractic/mycinema/config/SwaggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.context.annotation.Configuration;
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
+public class SwaggerConfig {
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/config/SwaggerConfig.java:13]: added swagger config

File: mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java [hunk 3]
```diff
@@ -15,66 +15,38 @@
     @Autowired
     PersonService personService;
 
-    @GetMapping(value = "/hello1")
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java:18]: clean unnecessary method

File: mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java [hunk 4]
```diff
@@ -15,66 +15,38 @@
...
-        return personService.helloFromService();
-    }
-
-    @PutMapping(value = "/saveMyPerson")
-    public Person saveMyPerson(@RequestBody Person personForSave) {
-        return personService.savePerson(personForSave);
-    }
-
-    @GetMapping(value = "/getById/{id}")
+    @GetMapping(value = "/{id}")
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java:23]: don't use action name in url

File: mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java [hunk 5]
```diff
@@ -15,66 +15,38 @@
...
     public Person modifyEmail(@RequestParam(value = "id") Long id, @RequestParam(value = "newEmail") String newEmail) {
-        Person p = personService.modifyEmail(id, newEmail);
-        return personService.savePerson(p);
+        return personService.modifyEmail(id, newEmail);
     }
 
-    @DeleteMapping(value = "/deleteById/{id}")
-    public Person deleteById(@PathVariable(value = "id") Long id) {
-        return personService.deleteById(id);
+    @DeleteMapping(value = "/{id}")
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java:48]: you can't return the person you just deleted

File: mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java [hunk 6]
```diff
@@ -15,66 +15,38 @@
...
-        Person p = personService.modifyName(id, newName);
-        return personService.savePerson(p);
+        return personService.modifyName(id, newName);
     }
 
-    @PostMapping(value = "/modifyPhone")
+    @PatchMapping(value = "/phone")
     public Person modifyPhone(@RequestParam(value = "id") Long id, @RequestParam(value = "newNumber") String newNumber) {
-        Person p = personService.modifyPhone(id, newNumber);
-        return personService.savePerson(p);
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/controllers/PersonController.java:67]: the save should be done in service

File: mycinema/src/main/java/ro/fiipractic/mycinema/services/impl/PersonServiceImpl.java [hunk 7]
```diff
@@ -39,37 +32,27 @@ public Person getById(Long id) {
 
     @Override
     public Person modifyName(Long id, String newName) {
-        Person p = personRepository.findById(id).orElse(null);
-        if (p == null)
-            throw new RuntimeException("Non-existent ID!");
+        Person p = personRepository.findById(id).orElseThrow(() -> new RuntimeException("Non-existent ID!"));
         p.setFullName(newName);
-        return p;
+        return personRepository.save(p);
```
Comment (IrinaGlodeanu) [inline mycinema/src/main/java/ro/fiipractic/mycinema/services/impl/PersonServiceImpl.java:37]: the save should be done in the service

**Review by Flendor — APPROVED** [review]

**Pull request merged**